src/styles/withStyles.d.ts

```diff
@@ -38,9 +38,13 @@ export interface StyledComponentProps<ClassKey extends string = string> {
   innerRef?: React.Ref<any>;
 }
 
+// Diff / Omit taken from https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-311923766
+export type Diff<T extends string, U extends string> = ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T];
+export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
```

Comment:
FYI, `Diff` and `Omit` are already defined in `index.d.ts`:

https://github.com/mui-org/material-ui/blob/29cbe2c8c1fbb1ac7341ffd6e499a56f6030bf86/src/index.d.ts#L43-L47